modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/exception/ModuleExceptionHandler.java

```diff
@@ -117,10 +118,11 @@ private Throwable getExceptionCause(Throwable throwable) {
     // For subclasses of ModuleException, we use it as it already contains additional information
     if (throwable.getClass().equals(ModuleException.class)) {
       return throwable.getCause() != null
-          ? throwable.getCause()
+          ? suppressIfPresent(throwable.getCause(), MessagingException.class)
```

Comment:
this doesn't look correct. You're doing this ONLY when the exception is a `ModuleException`. You're ignoring all subclasses (which is what's more frequently used).

Reply:
Added an additional suppression to cover the rest of the cases